File: POO-envios/C++/Lista/lista.cpp
#include <iostream>
#include "lista.h"

void Lista::insere(int value){
    if(nElementos < TAM){
        this->vetor[nElementos] = value;
        this->nElementos++;
    }
    else std::cout << "Número máximo de elementos atingidos" << std::endl;
}

void Lista::retira(){
    if(nElementos > 0){
        this->vetor[nElementos] = -1;
        this->nElementos--;
    }
    else std::cout << "Impossível retirar mais elementos" << std::endl;
}

int Lista::first(){
    if(this->vetor[0] != -1){
        return vetor[0];
    }
    else std::cout << "Lista vazia" << std::endl; return -1;
}

int Lista::last(){
    if(this->vetor[0] != -1){
        return vetor[nElementos - 1];
    }
    else std::cout << "Lista vazia" << std::endl; return -1;
}

int Lista::size(){
    return nElementos;
}

void Lista::destroy(){
    delete this->vetor;
}

File: POO-envios/Rust/lista/src/lista.rs
pub struct lista {
    nElementos: i32
    vetor: vec(i32)
}

impl lista {
    pub fn new() -> lista {
        lista {
            nElementos: 0;
            vetor: Vec::new();
        }
    }

    pub fn insere(&mut self, value: i32) {
        if nElementos < 20 {
            self.vetor[0] = value;
            self.nElementos = self.nElementos + 1;
        } else {
            println!("Número máximo de elementos atingidos");
        }
    }
    pub fn retira(&mut self) {
        if nElementos > 0 {
            self.vetor[0] = -1;
            self.nElementos = self.nElementos - 1;
        } else {
            println!("Impossível retirar mais elementos");
        }
    }
    pub fn first(&mut self) -> i32 {
        if self.vetor[0] != -1 {
            vetor[0]
        } else println!("Lista vazia"); -1
    }
    pub fn last(&mut self) -> i32 {
        if self.vetor[0] != -1 {
            vetor[self.nElementos - 1]
        } else println!("Lista vazia"); -1
    }
    pub fn size(&mut self) -> i32 {
        self.nElementos
    }
}